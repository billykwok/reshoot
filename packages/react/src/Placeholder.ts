import { css } from 'linaria';

import h from './h';

const asPlaceholder = css`
  width: 100%;
  height: 0;
`;

type Props = { color: string; aspectRatio: number };

const Placeholder = ({ color, aspectRatio }: Props) =>
  h('div', {
    className: asPlaceholder,
    style: {
      color,
      background: color,
      paddingBottom: aspectRatio ? aspectRatio + '%' : 0
    }
  });

export default Placeholder;
