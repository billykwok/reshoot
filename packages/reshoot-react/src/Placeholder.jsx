// @flow
import * as React from 'react';

type Props = { color: string, aspectRatio: number };

function Placeholder({ color, aspectRatio }: Props) {
  return (
    <div
      style={{
        width: '100%',
        height: 0,
        background: color,
        paddingBottom: aspectRatio ? aspectRatio + '%' : 0
      }}
    />
  );
}

export default React.memo<Props>(Placeholder, () => true);
