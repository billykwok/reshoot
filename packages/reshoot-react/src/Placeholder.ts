import { css } from 'linaria';

import h from './h';

const asPlaceholder = css`
  width: 100%;
  height: 0;
`;

type Props = { color: string; aspectRatio: number };

function Placeholder({ color, aspectRatio }: Props) {
  return h('div', {
    className: asPlaceholder,
    style: {
      color,
      background: color,
      paddingBottom: aspectRatio ? aspectRatio + '%' : 0
    }
  });
}

export default Placeholder;
