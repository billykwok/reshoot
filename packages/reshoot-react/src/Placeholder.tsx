import React from 'react';
import { jsx } from '@emotion/core';

type Props = { color: string; aspectRatio: number };

function Placeholder({ color, aspectRatio }: Props) {
  return jsx('div', {
    css: [
      { width: '100%', height: 0 },
      {
        color,
        background: color,
        paddingBottom: aspectRatio ? aspectRatio + '%' : 0
      }
    ]
  });
}

export default React.memo<Props>(Placeholder);
