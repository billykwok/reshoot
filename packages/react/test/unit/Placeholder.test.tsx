import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';

import Placeholder from '../../src/Placeholder';

describe('Placeholder tag', () => {
  test('is in good shape', () => {
    expect(
      renderer.create(<Placeholder aspectRatio={1.5} color="red" />).toJSON()
    ).toMatchSnapshot();
  });
});
