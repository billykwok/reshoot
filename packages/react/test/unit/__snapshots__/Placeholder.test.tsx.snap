// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Placeholder tag is in good shape 1`] = `
<div
  className="aund3wh"
  style={
    Object {
      "background": "red",
      "color": "red",
      "paddingBottom": "1.5%",
    }
  }
/>
`;
